### PR TITLE
navigate back

### PR DESCRIPTION
**Pull request by fuddl**
@Stvad for now, I want to show the navigation only if there is something to go back to. the sidebar might later also be holding a button that closes the overlay or moves it to the other side. the design is based on that of the navigation bar in iOS

demo

### CODE REVIEW

File: src/sidebar/prepend-nav.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { navigation } from './components/navigation/navigation.tpl.js';
+
+function PrependNav() {
+	document.body.insertBefore(navigation(), document.body.firstChild);
```
Comment (Stvad) [inline src/sidebar/prepend-nav.js:4]: can just use `prepend` https://developer.mozilla.org/en-US/docs/Web/API/Element/prepend ?

File: src/sidebar/components/navigation/navigation.tpl.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const navigation = () => {
+
+	let style = document.createElement('link');
+	style.setAttribute('rel', "stylesheet");
+	style.setAttribute('href', "components/navigation/navigation.css");
```
Comment (Stvad) [inline src/sidebar/components/navigation/navigation.tpl.js:5]: can you use the css import instead (e.g. see `ensign`) - it's less code and defines the dependency more explicitly

**Review by Stvad — COMMENTED** [review]
Nice! 
One thought I have is that the approach here requires to have controls injected in each page and does not account for when you accidentally navigate to a 3p url (e.g. link to a query or just random link).  

Both of those can be kind of sidestepped if you put controls on the level of iframe wrapper (e.g. the close button in #38 ), but I'm not sure if you can do that in the context of a native sidebar 🤔

**Comment by fuddl** [comment]
> Both of those can be kind of sidestepped if you put controls on the level of iframe wrapper (e.g. the close button in #38 ), but I'm not sure if you can do that in the context of a native sidebar 🤔 

I'd like to put as little code as possible into third party websites: so I would keep the controls _in_ the sidebar/overlay.

The native sidebar comes with it's own close button so I would only add in in the chrome overlay version.

**Pull request merged**